### ISSUE TITLE
p:ajax update inside p:columns breaks rendering

### ISSUE THREAD

**Issue by fabrii** [primefaces/primefaces#2577]
## 1) Environment
- PrimeFaces version: 6.1
- Application server + version: JBoss EAP 7.0
- Affected browsers: Chrome, Firefox

## 2) Expected behavior

Render every h:outputText output facet inside p:cellEditor inside p:columns.

## 3) Actual behavior

Only the first outputText (problem_output) is rendered when the table is loaded. The next columns render and empty outputtext. The method getByYear(year) is getting called and returns the proper entity.

The output facet are rendered:
 - If the outputText is just a plain text like this: <h:outputText id="problem_output" value="TEXT" />
 - If I click the cell and edit the inputNumber.

## 4) Sample XHTML

```xhtml

<p:dataTable  var="pre" value="#{MB.entity.entity2}" rowIndexVar="rowIndex"  editable="true" editMode="cell">
    <p:column headerText="#{msg.Field1}">
       <h:outputText value="#{pre.staticField}" />
    </p:column>

    <p:columns value="#{MB.entity.years}" var="year" columnIndexVar="colIndex">

        <f:facet name="header">
             <h:outputText value="Year #{year+1}" />
        </f:facet>

        <p:cellEditor>
            <f:facet name="output">
                <h:outputText id="problem_output" value="#{pre.getByYear(year).total}" />
            </f:facet>
            <f:facet name="input">
               <p:inputNumber value="#{pre.getByYear(year).total}"  decimalSeparator="." thousandSeparator=","/>
            </f:facet> 
        </p:cellEditor>
    </p:columns>

</p:dataTable>

```

UPDATE (31/07). Its not required to be inside p:columns. Just a p:cellEditor inside column with more than one column can reproduce the bug.

## 5) Sample bean

Years is a list like this: 0, 1, 2, 3.
GetByYear(year) returns an entity depending on the year.

## 6) Steps to reproduce

a ) Load table: 

Field1 | Year 1  | Year 2  | Year 3
------------ | ------------- | ------------- | -------------
Static Text |  |  | 
Static Text |  |  | 

b) Edit table, by using input facet. After that, the cells render the output properly:

Field1 | Year 1  | Year 2  | Year 3
------------ | ------------- | ------------- | -------------
Static Text | 15.000 | 6.000  | 4.000
Static Text | 12.000 | 5.000  | 4.000

Values are propagated to entity in MB.

c) After saving values and refreshing the page:

Field1 | Year 1  | Year 2  | Year 3
------------ | ------------- | ------------- | -------------
Static Text | 15.000 |  | 
Static Text | 12.000 |  | 

The last columns do not render the outputText value.

## 7) WORKAROUND

```javascript

<script>
      $(document).ready(function () {
          updateCellsOneByOne();
      })

      function updateCellsOneByOne(){
           forEach(idCell in theArray){
                PrimeFaces.ab({s: "", u: idCell});
            }
       }

</script>
```

**Comment by fabrii** [comment]
My workaround does not work if application is behind WAF. It detects multiple ajax calls and blocks the connection. Any thoughts?

**Comment by wal-jan** [comment]
@fabrii I cannot reproduce the behaviour you described.
[PF2577.zip](https://github.com/primefaces/primefaces/files/1435945/PF2577.zip)
 Please provide test case  (at least sample bean and "entity" classes)

**Comment by fabrii** [comment]
I was able to reproduce the bug if the <p:inputNumber> has a <p:ajax> inside like this:

<p:ajax event="change" 
                                        update="form:tabDatosProyecto:proyecto_presupuesto:#{rowIndex}:monto_total_rubro 
                                        form:second_accordion:#{colIndex}:table:#{rowIndex}:donaciones_porcentaje
                                        form:second_accordion:#{colIndex}:table:#{rowIndex}:contraparte_porcentaje
                                        form:second_accordion:#{colIndex}:table:#{rowIndex}:monto_total 
                                        form:third_accordion:#{colIndex}:table:#{rowIndex}:donaciones_porcentaje
                                        form:third_accordion:#{colIndex}:table:#{rowIndex}:contraparte_porcentaje
                                        form:third_accordion:#{colIndex}:table:#{rowIndex}:donaciones_desvio
                                        form:third_accordion:#{colIndex}:table:#{rowIndex}:contraparte_desvio
                                        form:third_accordion:#{colIndex}:table:#{rowIndex}:monto_total                                                    
                                        "/>

Ill try to reproduce it in your project and upload it.

**Comment by fabrii** [comment]
[PF2577.zip](https://github.com/primefaces/primefaces/files/1442815/PF2577.zip)
@wal-jan Here it is!

**Comment by wal-jan** [comment]
I just simplified Fabricios test case.  [PF2577.zip](https://github.com/primefaces/primefaces/files/1443572/PF2577.zip) So yes it's reproducible.

Looks like a rendering problem of `p:ajax`s `update` attribute if it contains `p:dataTable`s row index and/or `p:columns`s column index.

Maybe the title of this issue should be renamed to "p:ajax update inside p:columns breaks rendering" (or similar).

I observed that rendering works well if index variables are used in `p:ajax`s `listener` attribute (but that does not solve the problem).
@fabrii as workaround just use `f:ajax` `render` instead of `p:ajax` `update`
